src/node/stage/approved.rs

```diff
@@ -971,18 +1222,118 @@ impl Approved {
+            debug!("splitting with new_infos {:?}", new_infos);
+        }
+
         for info in new_infos {
             let participants: BTreeSet<_> = info.elder_ids().copied().collect();
             let _ = self.dkg_cache.insert(participants.clone(), info);
-            self.consensus_engine.vote_for(NetworkEvent {
-                payload: AccumulatingEvent::StartDkg(participants),
-                proof_share: None,
-            });
+
+            // In case this elder is not a participant, which indicates a split. Check whether got
+            // notified of the sibling DKG result.
+            if let Some(dkg_result) = self.sibling_dkg_result_cache.remove(&participants) {
+                let _ = self.handle_dkg_result_event(core, &participants, &dkg_result);
+                continue;
+            }
+
+            // In case all the current elders split into one side of the sub-section, triggering the
+            // sibling section's DKG process by send them an Initial DKG message.
+            let elder_ids: Vec<PublicId> =
+                self.shared_state.our_info().elder_ids().copied().collect();
+            if participants
+                .iter()
+                .all(|participant| !elder_ids.contains(participant))
+            {
+                trace!(
+                    "Triggering DKG process for sibling section during splitting {:?}",
+                    participants
+                );
+                // TODO: allow threshold to be configured.
+                let threshold = participants.len() * 2 / 3;
+                let message = DkgMessage::Initialization {
+                    // FIXME: will be counted as an extra vote
+                    key_gen_id: 0,
+                    m: threshold,
+                    n: participants.len(),
+                    member_list: participants.clone(),
+                };
+                let _ = self.broadcast_dkg_message(
+                    core,
+                    participants,
+                    self.consensus_engine.parsec_version(),
+                    message,
+                );
+            } else {
+                self.init_dkg_gen(core, participants, self.consensus_engine.parsec_version());
+            }
         }
 
         true
     }
 
+    fn init_dkg_gen(
+        &mut self,
+        core: &mut Core,
+        participants: BTreeSet<PublicId>,
+        parsec_version: u64,
+    ) {
+        if self
+            .key_gen_map
+            .contains_key(&(participants.clone(), parsec_version))
+            || self
+                .completed_dkg
+                .contains(&(participants.clone(), parsec_version))
+        {
+            trace!(
+                "already have key_gen of {:?}-{:?}",
+                parsec_version,
+                participants
+            );
+            self.check_dkg(core);
+            return;
+        }
+
+        // TODO: allow threshold to be configured.
+        let threshold = participants.len() * 2 / 3;
```

Comment:
...same here. Or maybe define a function for it: `fn dkg_threshold(num_participants: usize) -> usize`

Reply:
Or use the `quorum_count` (in `src/section/elders_info.rs`)  function we already have, just subtract one from the result?